### PR TITLE
WIP: Refactoring

### PR DESCRIPTION
**Pull request by wey1and**
What has been done? Why? What problem is being solved?

I didn't forget about

- [ ] Tests
- [ ] Changelog
- [ ] Documentation
- [ ] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Close #???

### CODE REVIEW

File: src/main/java/io/tarantool/driver/handlers/TarantoolResponseHandler.java [hunk 1]
```diff
@@ -46,9 +46,8 @@ protected void channelRead0(ChannelHandlerContext ctx, TarantoolResponse taranto
                         break;
                     case IPROTO_OK:
                         try {
-                            TarantoolOkResult okResult = new TarantoolOkResult(tarantoolResponse.getSyncId(),
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/handlers/TarantoolResponseHandler.java:49]: Don't get why it was removed. Would you mind elaborating about?

File: src/main/java/io/tarantool/driver/core/RequestFutureManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Keeps track of submitted requests, finishing them by timeout and allowing asynchronous request processing
+ *
+ * @author Alexey Kuzin
+ */
+public class RequestFutureManagerImpl implements RequestFutureManager {
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/core/RequestFutureManagerImpl.java:19]: nit: This change looks too academic. Do we really suppose to have many RequestFutureManager implementations?

File: src/main/java/io/tarantool/driver/core/connection/TarantoolConnectionFactory.java [hunk 3]
```diff
@@ -31,9 +32,9 @@
  */
 public class TarantoolConnectionFactory {
 
-    private final TarantoolClientConfig config;
-    private final Bootstrap bootstrap;
-    private final ScheduledExecutorService timeoutScheduler;
+    protected final TarantoolClientConfig config;
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/core/connection/TarantoolConnectionFactory.java:35]: Why is this now visible for subclasses, is there a reason?

**Review by akudiyar — CHANGES_REQUESTED** [review]

File: src/main/java/io/tarantool/driver/core/RequestFutureManagerImpl.java (hunk 2 above)
Comment (bitgorbovsky) [reply src/main/java/io/tarantool/driver/core/RequestFutureManagerImpl.java:19]: Yes, it's being used in another project, for implementation similar module for managing request, but some additional possibilities

File: src/main/java/io/tarantool/driver/core/connection/TarantoolConnectionFactory.java (hunk 3 above)
Comment (bitgorbovsky) [reply src/main/java/io/tarantool/driver/core/connection/TarantoolConnectionFactory.java:35]: The reason is usage these class fields in subclasses of this factory

**Comment by bitgorbovsky** [comment]
Closed in favor of https://github.com/tarantool/cartridge-java/pull/243

**Pull request closed without merging**